src/plugins/ui/evaluation_conditional_format.ts

```diff
@@ -484,11 +492,26 @@ export class EvaluationConditionalFormatPlugin extends UIPlugin {
           if (origin.sheetId === target.sheetId) {
             this.adaptRules(origin.sheetId, cf, [xc], toRemoveRange);
           } else {
-            this.adaptRules(target.sheetId, cf, [xc], []);
             this.adaptRules(origin.sheetId, cf, [], toRemoveRange);
+            const cfToCopyTo = this.getCFToCopyTo(target.sheetId, cf);
+            this.adaptRules(target.sheetId, cfToCopyTo, [xc], []);
           }
         }
       }
     }
   }
+
+  private getCFToCopyTo(targetSheetId: UID, originCF: ConditionalFormat): ConditionalFormat {
+    const cfInTarget = this.getters
+      .getConditionalFormats(targetSheetId)
+      .find((cf) => areConditionalFormatEquivalents(originCF, cf));
+
+    if (cfInTarget) {
+      return cfInTarget;
+    }
+
+    const cf = { ...originCF, id: this.uuidGenerator.uuidv4(), ranges: [] };
+    this.dispatch("ADD_CONDITIONAL_FORMAT", { cf, target: [], sheetId: targetSheetId });
```

Comment:
this method is called `getBrol` but actually contains a side-effect (dispatch).
I'd move the dispatch outside.

Reply:
The best part is that this dispatch is 100% useless, it is not dispatched because the allowDispatch prevent dispatching a CF with empty ranges :upside_down_face: